tools/Vis_PV.py

```diff
@@ -0,0 +1,286 @@
+        slice1.SliceType.Center = slice_Properties['Center_c']
+        slice1.SliceType.Radius = slice_Properties['Radius_c']
+        slice1.SliceType.Axis = slice_Properties['Axis']
+    #--Hide previous data display before filter--
+    Hide(Filter1, renderView1)
+    xDMFReader1_Display = Show(slice1,renderView1)
+    return renderView1, slice1, xDMFReader1_Display
+
+def apply_warp(var, xDMFReader1, renderView1):
+    '''
+    Add
+    '''
+    slice1 = Slice(Input=xDMFReader1)
+    slice1.SliceType = 'Plane'
+    slice1.SliceOffsetValues = [0.0]
+    slice1.SliceType.Origin = [3.141592653589793, 3.141592653589793, 3.141592653589793]
+    slice1.SliceType.Normal = [0.0, 0.0, 1.0]
+    #xDMFReader1_Display.ColorArrayName = ['POINTS', var]
+    warpByScalar1 = WarpByScalar(Input=slice1)
+    warpByScalar1.Scalars = ['POINTS', var]
+    Hide(xDMFReader1, renderView1)
+    xDMFReader1_Display = Show(warpByScalar1, renderView1)
+    return renderView1
+
+def save_images(renderView1, xDMFReader1):
+    '''
+    Saves single image or multiple images based on number of 
+    time steps in data. One image per time step saved
+    '''
+    time_steps = xDMFReader1.TimestepValues
+    renderView1.ViewSize = [1330, 670]
+    try:
+        n_steps = len(time_steps)
+    except:
+        n_steps = 1
+        time_steps = [time_steps]
+    if n_steps ==1:
+        SaveScreenshot('image.png', renderView1)
+    elif n_steps > 1:
+        anim = GetAnimationScene()
+        anim.PlayMode = 'Snap To TimeSteps'
+        for n in range(n_steps):
+            print('loop',n)
+            anim.AnimationTime = time_steps[n]
+            view = GetRenderView()
+            SaveScreenshot('image_%s.png' %n, view)
+    return None
+
+def main():
+    '''
+    Render image(s) based on parameters specified 
+    '''
+    # Get data from file, set default camera properties
+    Input_data = load_InputData()
+    renderView1, xDMFReader1, xDMFReader1_Display = readData_Xdmf(Input_data['file_path'])
+    set_defaultCamera(renderView1)
+    # variable name to 'var'
+    var = Input_data['Variable_properties']['Variable_name']
+
+    #renderView1 = tetrahedralize(xDMFReader1, renderView1)
+    # Apply filters
+    Filter1 = GetActiveSource()
+
+    if Input_data['Filter']['Clip']['Clip_type'] == Input_data['Filter']['Slice']['Slice_type']:
+        sys.exit('Clip type and Slice type chosen cannot be applied simulatneously')
+    else:
+        if Input_data['Filter']['Clip']['Apply'] == True:
+            renderView1, Filter1, xDMFReader1_Display = apply_clip(Input_data['Filter']['Clip'],
+                                                                   var, renderView1, Filter1)
+        elif Input_data['Filter']['Slice']['Apply'] == True:
+            renderView1, Filter1, xDMFReader1_Display = apply_slice(Input_data['Filter']['Slice'],
+                                                                    var, renderView1, Filter1)
+    renderView1.Update()
```

Comment:
Add a comment along the lines of `# Update outside of apply_clip and apply_slice so that both a clip and a slice could be done simultaneously in the future`